### ISSUE TITLE
Temporal horizon vs output confidence scaling

### ISSUE THREAD

**Issue by justinshenk** [justinshenk/temporal-awareness#18]
## Motivation

Do models become less confident when reasoning about distant futures? This experiment directly measures whether output confidence decreases with temporal distance.

Idea by Nicola Sabbadini.

## Research Question

Does aggregate token confidence scale inversely with temporal horizon?

### Setup

```python
horizons = ["next hour", "tomorrow", "next week", "next month", "next year", "next decade", "next century"]

# Generate matched questions across horizons
question_templates = [
    "What will the weather be like {horizon}?",
    "What will I be doing {horizon}?",
    "What will the stock market do {horizon}?",
    "What technology will exist {horizon}?",
]
```

### Method

1. Set temperature = 0 (greedy decoding)
2. For each question at each horizon:
   - Generate response
   - Extract logits for each token
   - Compute confidence = softmax probability of selected token
   - Aggregate: mean confidence across response tokens

```python
def measure_confidence(model, prompt):
    outputs = model.generate(prompt, temperature=0, return_logits=True)
    
    confidences = []
    for token_logits, selected_token in zip(outputs.logits, outputs.tokens):
        probs = softmax(token_logits)
        confidences.append(probs[selected_token])
    
    return {
        'mean_confidence': np.mean(confidences),
        'min_confidence': np.min(confidences),
        'confidence_std': np.std(confidences),
        'token_confidences': confidences
    }
```

### Analysis

Plot: X = time horizon (log scale), Y = aggregate confidence

Expected patterns:
- Decreasing confidence with horizon (uncertainty scaling)
- Or flat (model doesn't calibrate to temporal uncertainty)
- Or domain-dependent (weather vs technology)

### Extensions

1. Compare confidence scaling across models (GPT-2, Llama, Qwen)
2. Correlate with temporal probe activations
3. Segment by domain (predictable vs unpredictable futures)
4. Compare hedging language frequency vs actual confidence

## Success Criteria

| Metric | Target |
|--------|--------|
| Clear trend visualization | Interpretable graph |
| Statistical significance | p < 0.05 for horizon effect |
| Cross-model consistency | Same pattern across 2+ models |

## Deliverables

1. `scripts/experiments/confidence_scaling.py`
2. Visualization: confidence vs horizon graph
3. Statistical analysis of scaling relationship
4. Comparison with probe-detected uncertainty

## Connection to Other Issues

- Directly implements #6 (Uncertainty scaling) with logit-based measurement
- Complements #3 (Explicit temporal context) calibration analysis
- Informs #17 (Probe reliability) confidence calibration

## Track

Track B (Experiments - parallelizable)

## Thrust

Robustness & Failure Modes / Calibration
